privval/file.go

```diff
@@ -398,10 +398,20 @@ func (pv *FilePV) signVote(chainID string, vote *tmproto.Vote) error {
 	if err != nil {
 		return err
 	}
-	if err := pv.saveSigned(height, round, step, signBytes, sig); err != nil {
+	if err = pv.saveSigned(height, round, step, signBytes, sig); err != nil {
 		return err
 	}
 	vote.Signature = sig
+
+	// Sign the vote extension, if any
+	if len(vote.Extension) > 0 {
+		var err error
+		vote.ExtensionSignature, err = pv.Key.PrivKey.Sign(vote.Extension)
```

Comment:
I had missed this in my previous pass.
Here, we are signing the extension's bytes directly, but we should be constructing a `CanonicalVoteExtension` and then signing it. Check the call to `VoteSignBytes` earlier in this function... we should be doing something similar with the vote extension